### PR TITLE
MM-54278 - Calls banners redesign

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- This PR changes and does a refactor of the calls banner design and UI, partly for polish, partly to accommodate the upcoming bookmarks feature.
- It also refactors the internal design (to rely more on flex gap instead of manually computing distances and placing banners with absolute positions). This simplifies a lot of the code.
- The voice-on (talking) shadow has been replaced by the dual border effect for iOS. This makes it look like Android now.
- The talking borders are a bit smaller in the current call, a bit bigger in the call screen.
- Emojis were also inconsistent across devices and avatar sizes, made them similar. Also tightened up some of the spacing between the avatars and the names
- Todo:
  - adjust the New Messages bar for the new UI, and place it under all the banners (after adding the error bars and the incoming call notifications it was hidden).
  - call from another server changes
  - calls banners on the channel list screen

#### Designs
- https://www.figma.com/file/WVbRdxA4aRFdw410j2DxxU/MM-51513-Ringing-in-DMs%2C-GMs-(Mobile)?type=design&node-id=1378-95706&mode=design&t=CoxDrNZEd8wk9Y0B-0

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-53956

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 13, Pixel 6
- iOS: 16.5.1, iPhone 14

#### Screenshots

Android: (ios looks the same)

NOTE: the first two call quality banners have the correct styling, imagine the others have it too.

|col1|col2|col3|
|---|----|---|
| Screenshot_20231017-174950 | Screenshot_20231017-172024 | Screenshot_20231017-172032 |
| Screenshot_20231017-172055 | Screenshot_20231017-175132 | Screenshot_20231017-171547 |
| Screenshot_20231017-171431 | Screenshot_20231017-171409 | Screenshot_20231017-171339 |
| Screenshot_20231017-165254 | |

iOS so you can see that the emojis are a bit bigger now (matches Android), and the voice indicator is the same as Android.

| col1 | col2 |
|---|---|
| IMG_D899620474FB-1 |  IMG_85E8F01037AD-1 |

#### Release Note

```release-note
Calls: UI redesign and polish
```

### CODE REVIEW

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Great work @cpoile. You were very faithful to the designs :) I don't have much to comment on - just these few things. 

### All banners
- Are we able to add a shadow to the banners? Or is that not easy to pull off? Designs have a subtle shadow for all banners
- Also, I'd like to remove the borders on the banners - with the exception of the active call widget (that one needs it for dark themes). However, we can reduce the opacity on the border to 0.08.

### Active Call Widget
- For the 'no one is talking' state, I added a [screen in Figma](https://www.figma.com/file/WVbRdxA4aRFdw410j2DxxU/MM-51513-Ringing-in-DMs%2C-GMs-(Mobile)?type=design&node-id=1474-5588&mode=design&t=xHCELhvX8C16zlNx-4) that improves this a little. Could you update that while we're making changes here?
      image

### Incoming call banner
- As discussed in our DM, I seem to be getting an in-app notification overlaid on top of the incoming call
    image

### Call quality banner
- Looks like the `x` on the call quality banner has to update its color
    image
- Might need a _very_ slight adjustment for the icon and text to align with other banners. Or do my eyes deceive me?
    image
- Same goes for the red microphone banner

**Comment by cpoile** [comment]
@matthewbirtch Updated:
- light shadows (matching what we have for the New Messages banner, but I also added some for Android
- removed borders (except for Current call, and notifications in the Call Screen)
- made the changes for the 'no one is talking' state
- the push notification needs to be investigated: https://mattermost.atlassian.net/browse/MM-55036
- the banners should align now

| | |
|--|--|
| IMG_F966FD40DFA4-1 | IMG_A76FBDE68F8F-1 |
|  IMG_1326 |  IMG_1324 |

**Comment by matthewbirtch** [comment]
> @matthewbirtch Updated:
> 
> * light shadows (matching what we have for the New Messages banner, but I also added some for Android
> * removed borders (except for Current call, and notifications in the Call Screen)
> * made the changes for the 'no one is talking' state
> * the push notification needs to be investigated: https://mattermost.atlassian.net/browse/MM-55036
> * the banners should align now

Looks good @cpoile. Thanks for making those adjustments.

For the notification issue, I wanted to be clear that it's not actually a push notification that I'm seeing. It's an in-app notification. Those are used when you get notifications from other channels or DMs you're not currently viewing. So, for incoming calls, we should prevent those from showing. 

One other thing I just noticed in your screenshots is that the active call banner looks like it has too much padding on the left side now.

image

**Comment by enahum** [comment]
can you share a screenshot of the user avatar stack on the post list in a channel to see the impact on CRT posts ?

**Comment by cpoile** [comment]
@matthewbirtch Good eye, here's the fix (I think it matches; I'm going by the figma distances and using those as the margin/paddings).

| | |
|--|--|
| IMG_1329 | IMG_1330 |

@enahum Here's the pre (main) vs post (this pr):

| pre | post |
|--|--|
| IMG_FF963F2D6D39-1 | IMG_608042883F26-1 |

**Comment by matthewbirtch** [comment]
> @matthewbirtch Good eye, here's the fix (I think it matches; I'm going by the figma distances and using those as the margin/paddings).
> 
> IMG_1329	IMG_1330

👍 looks good now @cpoile. Were you going to look into the in-app notifications issue separately?

**Review by matthewbirtch — APPROVED** [review]

File: app/products/calls/components/incoming_calls_container.tsx [hunk 1]
```diff
@@ -3,63 +3,43 @@
...
-    showingCurrentCallBanner,
-    threadScreen,
 }: Props) => {
     const incomingCalls = useIncomingCalls().incomingCalls;
-    const insets = useSafeAreaInsets();
-    const micPermissionsGranted = useGlobalCallsState().micPermissionsGranted;
-    const currentCall = useCurrentCall();
+
+    // const micPermissionsGranted = useGlobalCallsState().micPermissionsGranted;
+    // const currentCall = useCurrentCall();
```
Comment (enahum) [inline app/products/calls/components/incoming_calls_container.tsx:27]: on purpose?

File: app/products/calls/components/incoming_calls_container.tsx [hunk 2]
```diff
@@ -3,63 +3,43 @@
...
-        (showingJoinCallBanner ? JOIN_CALL_BAR_HEIGHT : 0) +
-        (showingCurrentCallBanner ? CURRENT_CALL_BAR_HEIGHT - 2 : 0) +
-        (micPermissionsError ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
-        (qualityAlert ? CALL_ERROR_BAR_HEIGHT + 8 : 0);
-    const wrapperTop = {top};
+    // const micPermissionsError = !micPermissionsGranted && (currentCall ? !currentCall.micPermissionsErrorDismissed : false);
+    // const qualityAlert = (currentCall ? currentCall.callQualityAlert && currentCall.callQualityAlertDismissed === 0 : false);
+    // const marginTop = (micPermissionsError ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
+    //     (qualityAlert ? CALL_ERROR_BAR_HEIGHT + 8 : 0);
+    // const wrapperTop = {marginTop};
```
Comment (enahum) [inline app/products/calls/components/incoming_calls_container.tsx:39]: on purpose?

**Review by enahum — APPROVED** [review]

File: app/products/calls/components/incoming_calls_container.tsx (hunk 1 above)
Comment (cpoile) [reply app/products/calls/components/incoming_calls_container.tsx]: nope, by accident, thanks!

File: app/products/calls/components/incoming_calls_container.tsx (hunk 2 above)
Comment (cpoile) [reply app/products/calls/components/incoming_calls_container.tsx]: same, thanks

**Review by streamer45 — APPROVED** [review]
Very thorough, thanks :+1:

**Pull request merged**